### DISCUSSION TITLE
What can I do to progress to the point where I understand what you are doing here?

### DISCUSSION THREAD

**Discussion by rvencu** [lucidrains/DALLE-pytorch#224] [General]
Hi everyone,

Of course, I get the general idea but since I am recently getting myself into AI field some things are still out of my grasp.

I got an image classification project that I already solved with pretrained models such as Inception v4 but I am aiming to push the envelope and go beyond classification into image generation. I think this project may offer the best shot so I am struggling to bring myself up to speed to what is going on.

I would appreciate a list of resources for me to grasp the concepts then I would pay someone for a couple of mentoring sessions to nail it down.

Thanks.

**Comment by wolfgangmeyers** [comment]
I'd pay to join that mentoring session!

**Comment by chris-aeviator** [comment]
There's an excellent video deep-diving into how DALL-E works (seems to work to the Author of the video) --> https://www.youtube.com/watch?v=j4xgkjWlfL4

**Reply by rvencu** [reply]
Yes, and because of it, I arrived here... I am referring to a way to understand the components of the project (attention mechanism, losses, why a certain decision was made, etc). I know it is a lot but if someone can point me to "read this and this" it would be enough.

**Reply by Tollanador** [reply]
Hi rvencu,

Usually, the best way for people to learn, is just like these models, by _doing_ (for humans, though, it's playing). In other words, Just start playing with basic Autoencoders, That will give you an idea of the basics of image generation based on the training data. An easy place to start with that is Image Denoising. That will hopefully give you an understanding of what is going on (as opposed to just knowledge learned by reading).

By experiencing for yourself the pros and cons of basic autoencoders, you may begin to combine that info with your already learned knowledge. You may then think of potential ways to improve on the basic auto-encoder, then test those potentials .. you're basically aiming to _understand through play_,

The reason I suggest that you simply start by playing with these basic models first is that the whole deep learning space has exploded in publications, unpublished community experiments, article pieces, etc. There are so many good papers and articles to read and, as you know, such little precious time.

By playing with the models (like you did with image classification) you will hopefully run across problems that you want to solve, these problems are your guide to finding what papers/articles to read.

I found that my searches for how to solve X or Y problem very often resulted in articles found on 'towards data science', If you don't want to play with models and just want to read (which is fine, everyone learns best their own way) perhaps start browsing articles there.

OH, also, try not to limit your play to only image-based problems. Play with other modalities, then you can get a feel what parts of each type of model architecture is generalisable to others and which parts are more specific (then hopefully learn why that is).

Hope that helps

**Reply by rvencu** [reply]
Thanks for heads up :)
